src/nvim/edit.c

```diff
@@ -6066,27 +6066,30 @@ void free_last_insert(void)
 
 #endif
 
-/*
- * Add character "c" to buffer "s".  Escape the special meaning of K_SPECIAL
- * and CSI.  Handle multi-byte characters.
- * Returns a pointer to after the added bytes.
- */
+/// Add character "c" to buffer "s"
+///
+/// Escapes the special meaning of K_SPECIAL and CSI, handles multi-byte
+/// characters.
+///
+/// @param[in]  c  Character to add.
+/// @param[out]  s  Buffer to add to. Must have at least MB_MAXBYTES + 1 bytes.
+///
+/// @return Pointer to after the added bytes.
 char_u *add_char2buf(int c, char_u *s)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_WARN_UNUSED_RESULT
 {
   char_u temp[MB_MAXBYTES + 1];
```

Comment:
`utf_char2bytes()` needs a buffer of at most size `6` (actually `4` if we follow the Unicode standard), not 
`MB_MAXBYTES +1 (=22)`.